src/orion/core/worker/algo_wrappers/algo_wrapper.py

```diff
@@ -0,0 +1,148 @@
+""" ABC for a wrapper around an Algorithm. """
+from __future__ import annotations
+
+import copy
+from logging import getLogger as get_logger
+from typing import TYPE_CHECKING, Generic, Sequence, TypeVar
+
+from orion.algo.base import BaseAlgorithm
+from orion.algo.space import Space
+from orion.core.worker.trial import Trial
+
+if TYPE_CHECKING:
+    from orion.core.worker.experiment_config import ExperimentConfig
+
+logger = get_logger(__name__)
+
+AlgoT = TypeVar("AlgoT", bound=BaseAlgorithm)
+WrappedT = TypeVar("WrappedT", bound=BaseAlgorithm)
+
+# pylint: disable=too-many-public-methods
+class AlgoWrapper(BaseAlgorithm, Generic[AlgoT]):
+    """Base class for a Wrapper around an algorithm."""
+
+    def __init__(self, space: Space, algorithm: AlgoT):
+        super().__init__(space)
+        self._algorithm = algorithm
+
+    def suggest(self, num: int) -> list[Trial]:
+        trials = self.algorithm.suggest(num)
+        for trial in trials:
+            self.register(trial)
+        return trials
+
+    def observe(self, trials: list[Trial]) -> None:
+        for trial in trials:
+            self.register(trial)
+        self.algorithm.observe(trials)
+
+    @classmethod
+    def transform_space(cls, space: Space) -> Space:
+        """Transform an (outer) space, returning the (inner) space of the wrapped algorithm.
+
+        The output of this classmethod will be used as `self.algorithm.space`, when we create an
+        instance of `cls`.
+        """
+        return space
+
+    @property
+    def algorithm(self) -> AlgoT:
+        """Returns the wrapped algorithm."""
+        return self._algorithm
+
+    @property
+    def unwrapped(self):
+        """Returns the unwrapped algorithm."""
+        return self.algorithm.unwrapped
+
+    def unwrap(self, target_type: type[WrappedT]) -> WrappedT:
+        """Unwrap until the given type of wrapper or algorithm is encountered.
+
+        If it isn't, raises a RuntimeError.
+        """
+        node = self
+        while isinstance(node, AlgoWrapper):
+            if isinstance(node, target_type):
+                return node
+            node = node.algorithm
+        if isinstance(node, target_type):
+            return node
+        raise RuntimeError(
```

Comment:
This also could be a simple test to add.

Reply:
Should be fixed in cb616d2f22c915a7ce5c265e281b11f325ef04cc